File: server/src/routes/metricsRoutes.js
/**
 * Pseudocode:
 * 1) Define aggregated metrics endpoints.
 * 2) Lock to designer role.
 */

export const metricsRoutes = [
  { method: "GET", path: "/metrics/funnel", handlerName: "MetricsController.funnel", requiresRole: "designer" },
  {
    method: "GET",
    path: "/metrics/stage-stats",
    handlerName: "MetricsController.stageStats",
    requiresRole: "designer",
  },
  {
    method: "GET",
    path: "/metrics/progression",
    handlerName: "MetricsController.progression",
    requiresRole: "designer",
  },
  {
    method: "GET",
    path: "/metrics/fairness",
    handlerName: "MetricsController.fairness",
    requiresRole: "designer",
  },
  {
    method: "GET",
    path: "/metrics/compare",
    handlerName: "MetricsController.compare",
    requiresRole: "designer",
  },
];

File: server/src/routes/MetricsRoutes.java
package routes;

import java.util.List;
import types.Types;

/**
 * Pseudocode:
 * 1) Define aggregated metrics endpoints.
 * 2) Lock to designer role.
 */
public class MetricsRoutes {
  public static final List<Types.RouteDefinition> ROUTES = List.of(
    new Types.RouteDefinition(
      Types.HttpMethod.GET,
      "/metrics/funnel",
      "MetricsController.funnel",
      Types.Role.DESIGNER
    ),
    new Types.RouteDefinition(
      Types.HttpMethod.GET,
      "/metrics/stage-stats",
      "MetricsController.stageStats",
      Types.Role.DESIGNER
    ),
    new Types.RouteDefinition(
      Types.HttpMethod.GET,
      "/metrics/progression",
      "MetricsController.progression",
      Types.Role.DESIGNER
    ),
    new Types.RouteDefinition(
      Types.HttpMethod.GET,
      "/metrics/fairness",
      "MetricsController.fairness",
      Types.Role.DESIGNER
    ),
    new Types.RouteDefinition(
      Types.HttpMethod.GET,
      "/metrics/compare",
      "MetricsController.compare",
      Types.Role.DESIGNER
    )
  );
}
